### PR TITLE
Vector bounds

### PR DESCRIPTION
**Pull request by michellemho**
Current behavior:
- add `map.fitBounds()` function to map. Map loads on null island (default) and camera "flys" to the bounding box of the Layers (determined using `_get_bounds`)
- If only LocalLayers are passed to `ccmap`, the bounding box is the whole world
- Otherwise, if there are both Layers and LocalLayers passed to `ccmap`, only Layers are considered for the bounding box (Right now, there's no quick way to figure out the bounding box of a geojson source!)

### CODE REVIEW

**Comment by michellemho** [comment]
Noticed error when trying to add a polygon geojson as a LocalLayer from carto-vl.min.js:6:

`Uncaught (in promise) t.a: first polygon ring must be external.`

I think this might be a carto-vl problem.

**Comment by andy-esch** [comment]
🎉 thanks! I’ll take a look :)

File: cartoframes/contrib/vector.py [hunk 1]
```diff
@@ -8,14 +8,14 @@
...
 
-
 class QueryLayer:
     def __init__(self, query, color=None, size=None, time=None):
         self.query = query
         self.color = color
         self.size = size
         self.time = time
-
+        self.orig_query = query
```
Comment (andy-esch) [inline cartoframes/contrib/vector.py]: Why is this line needed?

File: cartoframes/contrib/vector.py [hunk 2]
```diff
@@ -8,14 +8,14 @@
...
+        self.is_basemap = False
```
Comment (andy-esch) [inline cartoframes/contrib/vector.py]: Same here -- is it needed or planned on being needed?

File: cartoframes/contrib/vector.py [hunk 3]
```diff
@@ -26,24 +26,28 @@ def __init__(self, query, color=None, size=None, time=None):
...
         basemap = 'DarkMatter'
     credentials = {} if creds is None else dict(user=creds.username(), api_key=creds.key())
+
+
     return (
         srcdoc\
             .replace('@@SOURCES@@', json.dumps(sources))
             .replace('@@BASEMAPSTYLE@@', basemap)
             .replace('@@CREDENTIALS@@', json.dumps(credentials))
+            .replace('@@BOUNDS@@', bounds)
```
Comment (andy-esch) [inline cartoframes/contrib/vector.py]: This should be a `json.dumps(bounds)` to ensure it's converted as we expect

File: cartoframes/contrib/vector.py (hunk 2 above)
Comment (michellemho) [reply cartoframes/contrib/vector.py]: Same story: for `_get_bounds()` function (found in context.py)

https://github.com/CartoDB/cartoframes/blob/2c5dec5c9bfe54aae78b4ee63b140ccb0588500b/cartoframes/context.py#L1772-L1774

File: cartoframes/contrib/vector.py (hunk 1 above)
Comment (michellemho) [reply cartoframes/contrib/vector.py]: I think this is needed so that I can use `_get_bounds()` function (found in context.py) 

https://github.com/CartoDB/cartoframes/blob/2c5dec5c9bfe54aae78b4ee63b140ccb0588500b/cartoframes/context.py#L1772-L1774

File: cartoframes/contrib/vector.py (hunk 3 above)
Comment (michellemho) [reply cartoframes/contrib/vector.py]: Okay, but I don't understand why? `bounds` is a string type object already (just like basemap)

**Comment by andy-esch** [comment]
The `Uncaught (in promise) t.a: first polygon ring must be external.` thing is a carto vl thing: https://github.com/CartoDB/carto-vl/issues/365
